### PR TITLE
Automatic Rubric Config Setup

### PR DESCRIPTION
**Pull request by mewilker**
## Overview

Resolves #650

This PR should cut back on required intial setup as it will setup the Rubric Config Database Table automatically if there is none detected.

## Details

- `RubricConfigDao` interface reuqires a method to setup default configs
- Memory and sql dao constructors call the setup default configs method
- Added unit test to confirm this
- changed memory dao unit test clear implementation
- changed `RubricConfig` equals method to consider null rubric items and missing rubric items the same
- SQL dao logs warnings when the config differs from the default (except for rubric id differences)
- updated documentation

## Testing

- [x] Added/updated unit tests
- [ ] Tested edge cases
- [ ] Manual testing (if needed)

## Dependencies

For #643, make sure to merge this and update the startup statements, and problably the documentation.

### CODE REVIEW

File: docs/getting-started/db-insert-statements/insert-rubric-database.md [hunk 1]
```diff
@@ -10,7 +10,12 @@ The primary things that needs to be updated on an ongoing basis are:
 
 ## Statements
 > [!NOTE]
-> Updated as of **WINTER 2026**
+> As of **SUMMER 2026** this step is no longer necessary. Rubric Configs are defined in the `RubricConfigDao` interface.
+upda> If you would like to make a change, please update the default configs there. **Live edits still require alter
```
Comment (TwoLettuce) [inline docs/getting-started/db-insert-statements/insert-rubric-database.md]: small typo here on line 14 

Do we still need this file if it's no longer a needed step? We should also probably remove references to it in the [getting started guide](https://github.com/softwareconstruction240/autograder/blob/main/docs/getting-started/getting-started.md) (step 7), or at least clarify there as well that it is no longer needed. also at the very end of the file

Comment (mewilker) [reply docs/getting-started/db-insert-statements/insert-rubric-database.md]: Spoke to Grant about this but documenting here:

The SQL statements can eventually be deleted, but I would like to keep them in for now as there are some changes Code coverage made to the descriptions. When #643 gets merged we can remove this file. There probably needs to be explanation to make sure rubric ids are kept up to date, but that documentation probably doesn't need it's own file.

**Review by TwoLettuce — APPROVED** [review]
Looks rad!

**Review by ZakkeryDaRebel — APPROVED** [review]
Looks good to me
